lib/resty/core/shdict.lua

```diff
@@ -372,22 +370,39 @@ local function shdict_incr(zone, key, value, init)
                 return error("bad init arg: number expected, got " .. typ)
             end
         end
+    end
 
-        has_init = 1
+    if init_ttl ~= nil then
+        local typ = type(init_ttl)
+        if typ ~= "number" then
+            init_ttl = tonumber(init_ttl)
+
+            if not init_ttl then
+                error("bad init_ttl arg: number expected, got " .. typ, 2)
+            end
+        end
+
+        if init_ttl < 0 then
+            error('bad "init_ttl" argument', 2)
+        end
+
+        if not init then
+            error('must provide "init" when providing "init_ttl"', 2)
+        end
 
     else
-        has_init = 0
-        init = 0
+        init_ttl = 0
     end
 
     local rc = C.ngx_http_lua_ffi_shdict_incr(zone, key, key_len, num_value,
-                                              errmsg, has_init, init,
+                                              errmsg, init and 1 or 0,
```

Comment:
OK, I now see why it used `has_init` as an integer in the first place...

Reply:
Yes... However, this patch still looks like an improvement to me though.

Reply:
@thibaultcha I didn't disagree :)

Reply:
😅